zenoh/src/net/routing/interceptor/authorization.rs

```diff
@@ -235,105 +346,148 @@ impl PolicyEnforcer {
                 validation_err.push_str("ACL config key_exprs list is empty. ");
             }
             if !validation_err.is_empty() {
-                bail!("{}", validation_err);
+                bail!("Rule '{}' is malformed: {}", config_rule.id, validation_err);
             }
-
-            // At least one must not be empty
-            let mut subject_validation_err: usize = 0;
-            validation_err = String::new();
-
-            if config_rule.interfaces.as_ref().unwrap().is_empty() {
-                subject_validation_err += 1;
-                validation_err.push_str("ACL config interfaces list is empty. ");
-            }
-            if config_rule.cert_common_names.as_ref().unwrap().is_empty() {
-                subject_validation_err += 1;
-                validation_err.push_str("ACL config certificate common names list is empty. ");
-            }
-            if config_rule.usernames.as_ref().unwrap().is_empty() {
-                subject_validation_err += 1;
-                validation_err.push_str("ACL config usernames list is empty. ");
+            for key_expr in config_rule.key_exprs.iter() {
+                if key_expr.trim().is_empty() {
+                    bail!("Found empty key expression in rule '{}'", config_rule.id);
+                }
             }
+            rule_map.insert(config_rule.id.clone(), config_rule);
+        }
 
-            if subject_validation_err == 3 {
-                bail!("{}", validation_err);
+        for config_subject in subjects.iter_mut() {
+            if subject_id_map.contains_key(&config_subject.id) {
+                bail!(
+                    "Subject id must be unique: id '{}' is repeated",
+                    config_subject.id
+                );
             }
-
-            for subject in config_rule.interfaces.as_ref().unwrap() {
-                if subject.trim().is_empty() {
-                    bail!("found an empty interface value in interfaces list");
+            // validate subject config fields
+            let interfaces = config_subject.interfaces.as_ref().unwrap();
```

Comment:
If the `unwrap` is not supposed to panic, I suggest using `expect` alongside the justification as to why. Same goes for the two `unwrap`s blow.